File: check-db.js
const sqlite3 = require('better-sqlite3');
const db = new sqlite3('data-scripts/processed/vocabulary.db');

console.log('=== 데이터 통계 ===');
console.log('총 단어 수:', db.prepare('SELECT COUNT(*) as count FROM words').get().count);
console.log('총 의미 수:', db.prepare('SELECT COUNT(*) as count FROM word_meanings').get().count);
console.log('총 예문 수:', db.prepare('SELECT COUNT(*) as count FROM examples').get().count);

console.log('\n=== 단어 샘플 ===');
const samples = db.prepare(`
  SELECT w.word, w.pronunciation, w.difficulty_level,
         wm.korean_meaning, wm.part_of_speech
  FROM words w
  LEFT JOIN word_meanings wm ON w.id = wm.word_id
  LIMIT 5
`).all();

samples.forEach(s => {
  console.log(`${s.word} [${s.pronunciation || 'N/A'}] (${s.part_of_speech || 'N/A'}) - ${s.korean_meaning || 'N/A'} (Level: ${s.difficulty_level || 'N/A'})`);
});

console.log('\n=== 난이도 분포 ===');
const levels = db.prepare('SELECT difficulty_level, COUNT(*) as count FROM words GROUP BY difficulty_level').all();
levels.forEach(l => console.log(`Level ${l.difficulty_level}: ${l.count}개`));

db.close();

File: check_db.py
import sqlite3
import sys

def check_word_meanings(db_path, word):
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    query = """
    SELECT w.word, wm.korean_meaning, wm.part_of_speech, wm.definition_en
    FROM words w
    LEFT JOIN word_meanings wm ON w.id = wm.word_id
    WHERE w.word = ?
    ORDER BY wm.id
    """

    cursor.execute(query, (word,))
    results = cursor.fetchall()

    print(f"\n=== '{word}' 단어의 모든 의미들 ===")
    if not results:
        print(f"'{word}' 단어를 찾을 수 없습니다.")
    else:
        for i, (word, korean, pos, english) in enumerate(results, 1):
            print(f"{i}. [{pos or 'N/A'}] {korean or 'N/A'}")
            if english:
                print(f"   영어 정의: {english}")
            print()

    conn.close()

if __name__ == "__main__":
    db_path = "data-scripts/processed/vocabulary.db"

    if len(sys.argv) > 1:
        word = sys.argv[1].lower()
        check_word_meanings(db_path, word)
    else:
        # 기본적으로 문제가 되는 단어들 확인
        problem_words = ['our', 'with', 'and', 'the', 'is', 'are']

        for word in problem_words:
            check_word_meanings(db_path, word)
            print("-" * 50)